app/cdap/components/hydrator/components/TopPanel/TopPanel.tsx

```diff
@@ -223,6 +230,18 @@ export const TopPanel = ({
   const exportDraftAndRedirect = () => {
     downloadFile(getConfigForExport());
     window.onbeforeunload = null;
+    // if there is a draft saved, delete it
+    if (stateParams.draftId) {
+      MyPipelineApi.deleteDraft({
+        context: getCurrentNamespace(),
+        draftId: stateParams.draftId,
+      }).subscribe({
```

Comment:
if this fails and you close the window would cdap be in a messed up state?

Reply:
No it's just for some reason you were not able to delete the draft. Worst case you have that extra draft in the draft list